tempo_embeddings/text/passage.py

```diff
@@ -138,42 +180,25 @@ def word_span(self, start, end) -> tuple[int, int]:
                 end,
                 self.text,
             )
-
         return self.tokenization.word_to_chars(word_index)
 
-    def words(self, use_tokenizer: bool = True) -> list[str]:
-        """Returns the words in the passage.
-
-        Args:
-            use_tokenizer: If True, uses the tokenizer to split the passage into words.
-                Otherwise, splits the passage by whitespace.
+    def words(self) -> list[str]:
+        """Returns the full words in the passage if word spans are known. Otherwise, splits the passage by whitespace.
 
         Returns:
-            An iterable of words in the passage.
+            An list of words in the passage.
         """
 
-        if use_tokenizer:
-            if self.tokenization is None:
-                raise RuntimeError("Passage has no tokenization.")
-
-            char_spans = [self.tokenization.word_to_chars(i) for i in self.tokenization.word_ids() if i is not None]
-            tokens = [self.text[sp[0]: sp[1]] for sp in sorted(set(char_spans))]
+        if self.full_word_spans is not None:
+            tokens = [self.text[sp[0]:sp[1]] for sp in self.full_word_spans]
```

Comment:
I prefer to avoid `if not ... else` conditions because double negation is hard to comprehend.
This one can be converted into
```python
if self.full_word_spans is None:
    tokens = [
        token.strip(string.punctuation).strip() for token in self._text.split()
    ]
    tokens = [token for token in tokens if len(token) > 1]
else:
    tokens = [self.text[sp[0]:sp[1]] for sp in self.full_word_spans]